src/components/data/data-utils.js

```diff
@@ -695,6 +696,82 @@ export const ModalToolButton = ({ icon, text, disabled, ...props }) => {
   ])
 }
 
+export const DeleteEntityColumnModal = ({ workspaceId: { namespace, name }, column: { entityType, attributeName }, onDismiss, onSuccess }) => {
+  const [deleting, setDeleting] = useState(false)
+  const [deleteConfirmation, setDeleteConfirmation] = useState('')
+
+  const signal = Utils.useCancellation()
+
+  const deleteColumn = async () => {
+    try {
+      setDeleting(true)
+      await Ajax(signal).Workspaces.workspace(namespace, name).deleteEntityColumn(entityType, attributeName)
+      onDismiss()
+      onSuccess()
+    } catch (e) {
+      reportError('Unable to modify column', e)
+      setDeleting(false)
+    }
+  }
+
+  return h(Modal, {
+    title: 'Delete Column',
+    onDismiss,
+    okButton: h(ButtonPrimary, {
+      onClick: deleteColumn,
+      tooltip: Utils.cond(
+        [_.toLower(deleteConfirmation) !== 'delete column', () => 'You must type the confirmation message'],
+        () => 'Delete Column')
+    }, 'Delete column')
+  }, [
+    div(['Are you sure you want to permanently delete the column ',
+      span({ style: { fontWeight: 600, wordBreak: 'break-word' } }, attributeName),
+      '?']),
+    div({
+      style: {
+        fontWeight: 500,
+        marginTop: '1rem'
+      }
+    }, 'This cannot be undone.'),
+    div({ style: { marginTop: '1rem' } }, [
+      label({ htmlFor: 'delete-column-confirmation' }, ['Please type \'Delete Column\' to continue:']),
+      h(TextInput, {
+        id: 'delete-column-confirmation',
+        placeholder: 'Delete Column',
+        value: deleteConfirmation,
+        onChange: setDeleteConfirmation
+      })
+    ]),
+    deleting && spinnerOverlay
+  ])
+}
+
+export const HeaderOptions = ({ field, onSort, isEntityName, beginDelete, children }) => {
+  const columnMenu = h(MenuTrigger, {
+    closeOnClick: true,
+    side: 'bottom',
+    content: h(Fragment, [
+      h(MenuButton, { onClick: () => onSort({ field, direction: 'asc' }) }, ['Sort Ascending']),
+      h(MenuButton, { onClick: () => onSort({ field, direction: 'desc' }) }, ['Sort Descending']),
+      !isEntityName && h(MenuButton, { onClick: beginDelete }, ['Delete column'])
+    ])
+  }, [
+    h(Link, { 'aria-label': 'Workflow menu', onClick: e => e.stopPropagation() }, [
+      icon('cardMenuIcon', {
+        size: 16
+      })
+    ])
+  ])
+
+  return h(IdContainer, [id => div({
+    style: { flex: 1, display: 'flex', alignItems: 'center', cursor: 'pointer', width: '100%', height: '100%' },
+    'aria-describedby': id
+  }, [
+    children,
+    div({ style: { marginRight: '1rem', marginLeft: 'auto' } }, [columnMenu])
```

Comment:
I'm surprised you need the marginLeft here. I think 'auto' is the default most of the time.

Reply:
Unfortunately, when I remove it, it does this:
<img width="526" alt="Screen Shot 2021-10-13 at 12 30 42 AM" src="https://user-images.githubusercontent.com/17534065/137067613-86d605bc-9a85-44fa-a0bd-274f74ad4faf.png">